lib/nerves_ssh/options.ex

```diff
@@ -258,27 +259,15 @@ defmodule NervesSSH.Options do
 
   defp resolve_system_dir(opts) do
     cond do
-      File.dir?(opts.system_dir) ->
+      File.dir?(opts.system_dir) or match?(:ok, File.mkdir_p(opts.system_dir)) ->
         opts
 
-      File.dir?(@default_system_dir) and host_keys_readable?(@default_system_dir) ->
-        %{opts | system_dir: @default_system_dir}
+      File.dir?(@default_host_system_dir) ->
+        %{opts | system_dir: @default_host_system_dir}
 
       true ->
-        %{opts | system_dir: :code.priv_dir(:nerves_ssh)}
-    end
-  end
-
-  defp host_keys_readable?(path) do
-    ["ssh_host_rsa_key", "ssh_host_dsa_key", "ssh_host_ecdsa_key"]
-    |> Enum.map(fn name -> Path.join(path, name) end)
-    |> Enum.any?(&readable?/1)
-  end
-
-  defp readable?(path) do
-    case File.read(path) do
-      {:ok, _} -> true
-      _ -> false
+        _ = File.mkdir_p(@default_device_system_dir)
```

Comment:
If this directory can't be created, it's because there's not a writable application directory. That's a valid Nerves configuration. We use it at SmartRent - I can show you. Since having ssh access is so critical, I think the keys should be put in `/tmp/nerves_ssh` and get created on every reboot. Unless I'm missing something, I think that in this case, it won't be possible to log in to the device if the application directory isn't writable.